hnn_core/basket.py

```diff
@@ -147,51 +127,37 @@ def parreceive_ext(self, type, gid, gid_dict, pos_dict, p_ext):
                     'type_src': type
                 }
 
-                # connections depend on location of input - why only
-                # for L2 basket and not L5 basket?
-                if p_ext['loc'] == 'proximal':
-                    self.ncfrom_ev.append(self.parconnect_from_src(
-                        gid_ev, nc_dict_ampa, self.soma_ampa))
-                    # NEW: note that default/original is 0 nmda weight for
-                    # the soma (for prox evoked)
-                    self.ncfrom_ev.append(self.parconnect_from_src(
-                        gid_ev, nc_dict_nmda, self.soma_nmda))
-
-                elif p_ext['loc'] == 'distal':
-                    self.ncfrom_ev.append(self.parconnect_from_src(
-                        gid_ev, nc_dict_ampa, self.soma_ampa))
-                    self.ncfrom_ev.append(self.parconnect_from_src(
-                        gid_ev, nc_dict_nmda, self.soma_nmda))
+                # NEW: note that default/original is 0 nmda weight
+                # for the soma (both prox and distal evoked)
+                for receptor in ['ampa', 'nmda']:
+                    self._connect_feed_at_loc(
+                        feed_loc='proximal', receptor=receptor,
+                        gid_src=gid_ev, nc_dict=nc_dict[receptor],
+                        nc_list=self.ncfrom_ev)
 
         elif type == 'extgauss':
             # gid is this cell's gid
             # gid_dict is the whole dictionary, including the gids
             # of the extgauss
-            # pos_list is also the pos of the extgauss (net origin)
+            # pos_dict is also the pos of the extgauss (net origin)
             # p_ext_gauss are the params (strength, etc.)
-            # I recognize this is ugly (hack)
             if self.celltype in p_ext.keys():
-                # since gid ids are unique, then these will all be shifted.
-                # if order of extgauss random feeds ever matters (likely)
-                # then will have to preserve order
-                # of creation based on gid ids of the cells
-                # this is a dumb place to put this information
                 gid_extgauss = gid + gid_dict['extgauss'][0]
 
-                # gid works here because there are as many pos
-                # items in pos_dict['extgauss'] as there are cells
                 nc_dict = {
                     'pos_src': pos_dict['extgauss'][gid],
                     # index 0 is ampa weight
                     'A_weight': p_ext[self.celltype][0],
-                    'A_delay': p_ext[self.celltype][1],  # index 2 is delay
+                    'A_delay': p_ext[self.celltype][2],  # index 2 is delay
```

Comment:
So was this a bug?

Reply:
interesting, I didn't see this myself. I guess it popped out in the diff when I consolidated the methods from the two classes. It's why we sorely need tests, thanks for pushing me on that. I'll try to make a test for that method.

Reply:
@blakecaldwell you might want to update it in HNN as well: https://github.com/jonescompneurolab/hnn/blob/master/L2_basket.py#L187

Reply:
Thanks @jasmainak